### DISCUSSION TITLE
TextBox在3.4--3.3版本中不同

### DISCUSSION THREAD

**Discussion by quhx** [HandyOrg/HandyControl#1352] [Q&A]
**3.4版本中TextBox没有实现IDataInput（图一示），3.3版本中TextBox实现了IDataInput（图二示）；这就导致3.4版本中无法使用一些属性（图三示）；为什么在3.4版本中去掉了TextBox的IDataInput实现**
![20230215165104](https://user-images.githubusercontent.com/66664939/218979290-da4853a2-5b50-4ea3-a141-6b60d5b7bba6.png)
![20230215165114](https://user-images.githubusercontent.com/66664939/218979311-1a55cde2-8a1f-458c-9d20-86f9ec63138f.png)
![2023-02-15_16-52-33](https://user-images.githubusercontent.com/66664939/218979596-5db64f57-1c3b-4d68-88fa-e1d412000672.png)

**Comment by NaBian** [comment]
已改为使用 ValidationRule 实现上述功能。
